### PR TITLE
Cancelling invocations

### LINKED ISSUE #52: Cancellations
This issue includes:

- https://github.com/restatedev/service-protocol/pull/50
- ~~Introduce `cancelled`/`failed` field for background invoke and resolve awakeable~~
- ~~Introduce failed completion variant for GetState~~
- ~~Introduce `StartMessage.cancel` field~~
- ~~Describe the SDK cancellation behavior in the protocol spec~~

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #52

### CODE REVIEW

File: service-invocation-protocol.md [hunk 1]
```diff
@@ -41,7 +41,8 @@ Depending on the specific syscall, the Restate runtime generates as response eit
 
 Each syscall defines a priori whether it replies with an ack or a completion, or doesn't reply at all.
 
-There are a couple of special message streams for initializing and closing the invocation.
```
Comment (slinkydeveloper) [inline service-invocation-protocol.md:44]: This text was unclear/wrong

File: service-invocation-protocol.md [hunk 2]
```diff
@@ -41,7 +41,8 @@ Depending on the specific syscall, the Restate runtime generates as response eit
...
+Depending on the programming language and/or specific SDK, the SDKs expose a way to await completions using some form of
```
Comment (slinkydeveloper) [inline service-invocation-protocol.md:44]: This clarification helps understand a bit the rest of the text i believe.

File: service-invocation-protocol.md [hunk 3]
```diff
@@ -345,6 +358,25 @@ closing the stream afterward.
...
+to the SDKs to decide how and when to surface the cancellation signal to the user code, be it an exception, or a
+saga-like library, or a mix of both. During the handling of the cancellation, the SDK MUST be able to suspend.
+
+The SDK must guarantee that the cancellation signal is fired in the user code deterministically, meaning that on a
+subsequent replay of the invocation, e.g. due to a suspension or a network error, the signal should be fired again on
+the same point. For this purpose the SDKs can record, when needed, the point where the cancellation signal was fired by
+using the `CancelledEntryMessage`.
+
+When the SDK doesn't implement cancellation, the runtime cancellation signal has no effect on the target invocation.
+
```
Comment (slinkydeveloper) [inline service-invocation-protocol.md:379]: Given the SDK has more or less freedom regarding how to implement the cancellation, I kept this text very generic, as from the protocol POV how the sdk exposes the cancellation signal and guarantees its determinism is not important.

Perhaps if we wanna describe these implementation details of the SDKs, we need some other form of text, like SDK features specs or similar.

**Review by tillrohrmann — COMMENTED** [review]
Based on the latest discussion about cancellations, I believe that this PR is currently not needed and can be closed.

**Pull request closed without merging**